internal/fs/fs.go

```diff
@@ -2077,13 +2077,30 @@ func (fs *fileSystem) WriteFile(
 	defer in.Unlock()
 
 	// Serve the request.
-	if err := in.Write(ctx, op.Data, op.Offset); err != nil {
-		return err
+	if fs.shouldTriggerBufferEnabledWrite(in) {
+		// Trigger write file buffering flow.
+		err := in.WriteToBuffer(fs.mountConfig.WriteConfig.BufferSizeMB, op.Data, op.Offset)
+		if err != nil {
+			return err
+		}
+	} else {
+		// Trigger the temp file flow.
+		if err := in.Write(ctx, op.Data, op.Offset); err != nil {
+			return err
+		}
 	}
 
 	return
 }
 
+func (fs *fileSystem) shouldTriggerBufferEnabledWrite(in *inode.FileInode) bool {
+	if fs.mountConfig.WriteConfig.EnableStreamingWrites &&
```

Comment:
`return fs.mountConfig.WriteConfig.EnableStreamingWrites &&
		in.IsLocal()`

instead of the if-else.